src/fedora/34/amd64/Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM registry.fedoraproject.org/fedora:34
+
+# Install the base toolchain we need to build anything (clang, cmake, make and the like)
+# this does not include libraries that we need to compile different projects, we'd like
+# them in a different layer.
+RUN dnf install -y \
+        clang \
+        cmake \
+        dnf-plugins-core \
+        findutils \
+        gdb \
+        glibc-langpack-en \
+        lldb-devel \
+        llvm-devel \
+        make \
+        python \
+        which \
+    && dnf clean all
+
+# Add MS package repo
+COPY  microsoft.asc /tmp
+RUN dnf config-manager --add-repo https://packages.microsoft.com/fedora/34/prod/ && \
+    rpm --import /tmp/microsoft.asc && \
+    rm /tmp/microsoft.asc
+
+# Install tools used by build automation.
+RUN dnf install -y \
+        git \
+        tar \
+        procps \
+        zip \
+    && dnf clean all
+
+# Dependencies of CoreCLR, Mono and CoreFX.
+RUN dnf install -y \
+        autoconf \
+        automake \
+        glibc-locale-source \
+        iputils \
+        krb5-devel \
+        libcurl-devel \
+        libgdiplus \
+        libicu-devel \
+        libtool \
+        libunwind-devel \
+        libuuid-devel \
+        lttng-ust-devel \
+        openssl-devel \
+        uuid-devel \
```

Comment:
It's a surprising that we need both `libuuid-devel` and `uuid-devel`. Is that intentional?

Reply:
I don't know. This is straight copy from existing 32. Also I don't know if we use it to actually build CoreCLR and others. It _feels_  like we could go away with just Helix & runtime dependencies.  But I did not want to fiddle with all. 
I really wanted updated Quic. The upgrade to Fedora 34 is opportunistic.